src/contexts/PreferencesContext.tsx

```diff
@@ -175,10 +178,10 @@ export const PreferencesProvider = ({
 
     (async function () {
       const backendPrefs = await fetchPreferences('zone');
-      const zoneArray = backendPrefs.map((pref: ZonePreference) => {
+      const zoneArray = backendPrefs?.map((pref: ZonePreference) => {
```

Comment:
I ran into an error at some point with this, and this along with the `|| []` resolved it, so I put it in each block.